docs/subscriptions_orders.md

```diff
@@ -0,0 +1,401 @@
+
+| Date Type | Description | 
+|-----------|-------------|
+| `start` | When the subscription begins |
+| `trial_end` | When the trial period ends |
+| `next_payment` | When the next payment is due |
+| `last_payment` | When the last payment was processed |
+| `end` | When the subscription is set to end |
+| `cancelled` | When the subscription was cancelled |
+| `payment_retry` | When the system will retry a failed payment |
+
+Each of these dates can be accessed via:
+
+```php
+$subscription->get_date('next_payment');
+```
+
+Or set via:
+
+```php
+$subscription->update_dates(array('next_payment' => '2023-12-01'));
+```
+
+## Order Relationships
+
+### Relationship Types
+
+WooCommerce Subscriptions defines several types of relationships between orders and subscriptions:
+
+```php
+private static $relation_types = array(
+    'renewal',
+    'switch',
+    'resubscribe',
+);
+```
+
+These relationship types define how orders relate to each other and to subscriptions.
+
+### How Relationships Are Stored
+
+The relationships between orders and subscriptions are stored using meta data on the orders:
+
+| Relationship | Meta Key | Description |
+|--------------|----------|-------------|
+| Parent Order → Subscription | `_subscription_id` | Links a parent order to its subscription(s) |
+| Renewal Order → Subscription | `_subscription_renewal` | Links a renewal order to its subscription |
+| Switch Order → Subscription | `_subscription_switch` | Links a switch order to its subscription |
+| Resubscribe Order → Subscription | `_subscription_resubscribe` | Links a resubscribe order to its subscription |
+
+### Relationship Management
+
+Relationships are managed through the `WCS_Related_Order_Store` class and its implementations. The primary methods for managing these relationships are:
+
+```php
+// Get related orders for a subscription
+$order_ids = WCS_Related_Order_Store::instance()->get_related_order_ids($subscription, 'renewal');
+
+// Get subscriptions related to an order
+$subscription_ids = WCS_Related_Order_Store::instance()->get_related_subscription_ids($order, 'renewal');
+
+// Add a relationship
+WCS_Related_Order_Store::instance()->add_relation($order, $subscription, 'renewal');
+
+// Remove a relationship
+WCS_Related_Order_Store::instance()->delete_relation($order, $subscription, 'renewal');
+```
+
+For performance, there's a cached implementation (`WCS_Related_Order_Store_Cached_CPT`) that maintains cache data on the subscription object.
+
+## Order Types in Detail
+
+### Parent Order
+
+**Definition**: The initial order that created the subscription.
+
+**Characteristics**:
+- Standard `shop_order` type
+- Contains subscription products that triggered subscription creation
+- May contain both subscription and non-subscription products
+- Processed like a normal WooCommerce order
+
+**How to identify**:
+```php
+// Check if an order is a parent order
+$is_parent = wcs_order_contains_subscription($order);
```

Comment:
Could be a little misleading. Might this be better?

```suggestion
$is_parent = wcs_order_contains_subscription( $order, 'parent' );
```

Otherwise `$is_parent` will be `true` even if the `$order` is a switch order.